tests/virtctl/vnc.go

```diff
@@ -0,0 +1,179 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+package virtctl
+
+import (
+	"context"
+	"fmt"
+	"image"
+	"image/png"
+	"net"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/mitchellh/go-vnc"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	launcherApi "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[rfe_id:127][crit:medium][arm64][vendor:cnv-qe@redhat.com][level:component][sig-compute][virtctl]VNC", decorators.SigCompute, func() {
+	var vmi *v1.VirtualMachineInstance
+
+	Describe("[rfe_id:127][crit:medium][vendor:cnv-qe@redhat.com][level:component]A new VirtualMachineInstance", func() {
+		BeforeEach(func() {
+			var err error
+			vmi = libvmifact.NewGuestless()
+			vmi, err = kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			vmi = libwait.WaitForSuccessfulVMIStart(vmi)
+		})
+
+		It("[test_id:5274]should connect to vnc with --proxy-only flag to the specified port", func() {
+			testPort := "33333"
+
+			By("Invoking virtctl vnc with --proxy-only")
+			proxyOnlyCommand := clientcmd.NewVirtctlCommand("vnc", "--proxy-only", "--port", testPort, "--namespace", vmi.Namespace, vmi.Name)
+
+			// Run this as go routine to keep proxy open in the background
+			go func() {
+				defer GinkgoRecover()
+				Expect(proxyOnlyCommand.Execute()).ToNot(HaveOccurred())
+			}()
```

Comment:
Not that you would be changing this but do we ever kill this command?

Reply:
That is a good question. The proxy command exits when the client that connected to the local port closes the connection. 

To enforce this I was thinking of a select statement and a done channel, but I don't see an easy way to implement this, since `Execute` is not returning a channel.

Reply:
The command has a context and you can get or actually set it. The problem is that vnc command needs to use this context. It should be trivial.

Reply:
I'd prefer to do it in a follow up PR. This requires changes and clean ups in the command itself too.

Reply:
Ok, let's open an issue for that.

Reply:
https://github.com/kubevirt/kubevirt/issues/13414